pipelines/zendesk/helpers/api_helpers.py

```diff
@@ -0,0 +1,99 @@
+    # oauth token is the preferred way to authenticate, followed by api token and then email + password combo
+    # if none of these are filled, simply return an error
+    if isinstance(credentials, ZendeskCredentialsOAuth):
+        zendesk_client = Zenpy(
+            subdomain=credentials.subdomain, oauth_token=credentials.oauth_token,
+            domain=domain, timeout=timeout, ratelimit_budget=ratelimit_budget, proactive_ratelimit=proactive_ratelimit, proactive_ratelimit_request_interval=proactive_ratelimit_request_interval,
+            disable_cache=disable_cache
+        )
+    elif isinstance(credentials, ZendeskCredentialsToken):
+        zendesk_client = Zenpy(
+            token=credentials.token, email=credentials.email, subdomain=credentials.subdomain,
+            domain=domain, timeout=timeout, ratelimit_budget=ratelimit_budget, proactive_ratelimit=proactive_ratelimit, proactive_ratelimit_request_interval=proactive_ratelimit_request_interval,
+            disable_cache=disable_cache
+        )
+    elif isinstance(credentials, ZendeskCredentialsEmailPass):
+        zendesk_client = Zenpy(
+            email=credentials.email, subdomain=credentials.subdomain, password=credentials.password,
+            domain=domain, timeout=timeout, ratelimit_budget=ratelimit_budget, proactive_ratelimit=proactive_ratelimit, proactive_ratelimit_request_interval=proactive_ratelimit_request_interval,
+            disable_cache=disable_cache
+        )
+    return zendesk_client
+
+
+def process_ticket(ticket: Ticket, custom_fields: DictStrStr, pivot_fields: bool = True) -> DictStrAny:
+    """
+    Helper that returns a dictionary of the ticket class provided as a parameter. This is done since to_dict()
+    method of Ticket class doesn't return all the required information
+    @:param sample_ticket: Ticket Object returned by a Zenpy API call, contains info on a single ticket.
+    @:param custom_fields: A dict containing all the custom fields available for tickets.
+    @:param pivot_fields: Bool that indicates whether to pivot all custom fields or not.
+    @:return base_dict: A dict containing 'cleaned' data about a ticket.
+    """
+
+    base_dict: DictStrAny = ticket.to_dict()
+    # get tags as a string eliminating the square brackets from string
+    base_dict["tags"] = str(base_dict["tags"])[1:-1]
+
+    # pivot custom field if indicated as such
+    # get custom fields
+    for custom_field in base_dict["custom_fields"]:
+        if pivot_fields:
+            cus_field_id = str(custom_field["id"])
+            field_name = custom_fields[cus_field_id]
+            base_dict[field_name] = custom_field["value"]
+        else:
+            custom_field["ticket_id"] = ticket.id
+
+    # delete fields that are not needed for pivoting
+    if pivot_fields:
+        del base_dict["custom_fields"]
+    else:
+        # un pivoting will simply save the dict as a json string
+        base_dict["custom_fields"] = str(base_dict["custom_fields"])
```

Comment:
do the same trick on `custom_fields` as with `tags`. you'll get a nice queryable JSON in the destination :)